### PR TITLE
223150: Nginx service flux config

### PR DESCRIPTION
**Pull request by ppotturi**
#Description
Added configuration to deploy the NGINX Plus using Flux.

#AB223150

### CODE REVIEW

File: infra/snd/01/nginx-ingress-values.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  namespace: config-cluster-flux
+spec:
+  chart:
+    spec:
+      version: 0.18.1
+  values:
+    controller:
+      image:
+        repository: "ssvcdoinfcr3401.azurecr.io/image/nginx-plus-ingress"  #Explore best way to set the repository name.
+        tag: 3.2.0
```
Comment (DaRoszko) [inline infra/snd/01/nginx-ingress-values.yaml:14]: 3.2.1 ?

File: core/nginx-ingress/repository.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: HelmRepository
+metadata:
+  name: repository-nginx-ingress
+  namespace: config-cluster-flux
+spec:
+  interval: 10m0s
```
Comment (DaRoszko) [inline core/nginx-ingress/repository.yaml:7]: what does the 10 minutes interview mean here exactly

**Comment by DaRoszko** [comment]
> 

I am generally happy with this PR, but would like some stuff added to the readme to detail what this repos generally about and what it contains @ppotturi  @kbabigumira

File: clusters/snd/01/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+apiVersion: kustomize.toolkit.fluxcd.io/v1
```
Comment (DaRoszko) [inline clusters/snd/01/kustomization.yaml:1]: I am generally happy with this PR, but would like some stuff added to the readme to detail what this repos generally about and what it contains @ppotturi @kbabigumira

**Review by DaRoszko — CHANGES_REQUESTED** [review]

File: core/nginx-ingress/release.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  chart:
+    spec:
+      chart: nginx-ingress
+      reconcileStrategy: ChartVersion
+      sourceRef:
+        kind: HelmRepository
+        name: nginx-ingress
+  install:
+    createNamespace: true
+  interval: 1m0s
```
Comment (DaRoszko) [inline core/nginx-ingress/release.yaml:16]: what this interview?

File: core/nginx-ingress/repository.yaml (hunk 2 above)
Comment (ppotturi) [reply core/nginx-ingress/repository.yaml:7]: it is to fetch the changes from nginx helm,  I have defaulted to 10mins. I think we need to refine all the intervals.

File: infra/snd/01/nginx-ingress-values.yaml (hunk 1 above)
Comment (ppotturi) [reply infra/snd/01/nginx-ingress-values.yaml:14]: Updated

File: core/nginx-ingress/release.yaml (hunk 4 above)
Comment (ppotturi) [reply core/nginx-ingress/release.yaml:16]: its the release reconciler interval.

File: clusters/snd/01/kustomization.yaml (hunk 3 above)
Comment (ppotturi) [reply clusters/snd/01/kustomization.yaml:1]: Readme updated

File: core/nginx-ingress/repository.yaml (hunk 2 above)
Comment (DaRoszko) [reply core/nginx-ingress/repository.yaml:7]: Change to 5m and we'll review later. any worries with that?

Comment (ppotturi) [reply core/nginx-ingress/repository.yaml:7]: This is to monitor ingress helm chart repro, I think we don't to monitor very often becasue we only looking for chat versions. I was thiking to chage to 24hs.

File: core/nginx-ingress/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+apiVersion: kustomize.toolkit.fluxcd.io/v1
```
Comment (kbabigumira) [inline core/nginx-ingress/kustomization.yaml:1]: Change this to kustomize.config.k8s.io/v1beta1

Comment (ppotturi) [reply core/nginx-ingress/kustomization.yaml:1]: We wanted to use flux one right?

File: README.md [hunk 6]
```diff
@@ -1,2 +1,33 @@
...
+
+## Repository Structure
+
+* `clusters` - Flux configuration for the Flux System (controllers etc) per environment
+* `infra` - Flux configuration for the core services e.g. Nginx Plus per environment
+* `core` - Contains the manifests for the core service (HelmRelease, HelmRepository)
+```
+
+├── clusters
+│   ├── snd
```
Comment (DaRoszko) [inline README.md:15]: is it worthwhile doing snd1, tst1, pre1, prd1, dev1, snd2, snd3 etc?

in docs & folder names - to match Azure env names exactly @kbabigumira  @ppotturi

Comment (DaRoszko) [reply README.md:15]: clusters/snd1/01 and clusters/snd2/01 ?
 
 I.e. within a subscription, we could have 2 AKS Clusters (01 and 02), but that would still be SND1 or PRD1, for example.

Comment (ppotturi) [reply README.md:15]: Are we going to have PRD1 and PRD2 environments? I think the PRD environment can have 10 and 02 clusters.

Comment (ppotturi) [reply README.md:15]: reason for snd folder is, all SND (1,2& 3) environments can always be the same config.

**Review by DaRoszko — DISMISSED** [review]

File: README.md (hunk 6 above)
Comment (DaRoszko) [reply README.md:15]: Design decision. Do we need different SND or TST env configurations? Wouldn't cluster 01 in SND1 have an update that SND2 Cluster 01 wouldn't ? for a bit?

Comment (DaRoszko) [reply README.md:15]: Environments can have the same config, but could they be different? are we differentiating SND1 2 and 3 via cluster number only?

File: core/nginx-ingress/repository.yaml (hunk 2 above)
Comment (kbabigumira) [reply core/nginx-ingress/repository.yaml]: @ppotturi  @DaRoszko  let's change this to a longer duration e.g. 24hrs

File: README.md (hunk 6 above)
Comment (ppotturi) [reply README.md:15]: As discussed, we are keeping the folder structure as it is.

**Review by DaRoszko — APPROVED** [review]

**Pull request merged**